### PR TITLE
copy only the libraries that were built

### PR DESCRIPTION
**Pull request by mightyguava**
As in https://github.com/bbqsrc/cargo-ndk/issues/89, we build multiple ffi libraries in the same cargo workspace. `cargo-ndk` ends up copying all libraries in the target directory instead of only the ones built for that invocation. Gradle then will complain with something like `2 files found with path 'lib/arm64-v8a/libffi.so' from inputs` in the merge JNI libraries step.

This change parses the library name from Cargo.toml and copies only the library (per target) to the output directory. I'm not sure if this will end up missing some libraries in copy. As far as I can tell, every rust crate can have 0 or 1 libraries, and `package.name` is required in `Cargo.toml` so we always have a name.

The library name is derived from `lib.name` if set, or from `package.name` if not, per https://doc.rust-lang.org/cargo/reference/cargo-targets.html#the-name-field

### CODE REVIEW

**Comment by mightyguava** [comment]
@bbqsrc how do I get this new change? Do I wait for you to get a new tag? Not sure how cargo plugins versioning works

**Comment by bbqsrc** [comment]
Published in 3.5.5. 😄

File: src/meta.rs [hunk 1]
```diff
@@ -16,11 +16,13 @@ fn default_targets() -> Vec<Target> {
 
 #[derive(Debug, Deserialize)]
 struct CargoToml {
-    package: Option<Package>,
+    package: Package,
```
Comment (MarijnS95) [inline src/meta.rs:19]: This change is breaking the parsing of `Cargo.toml` files which define a `[workspace]`, without defining a `[package]`: https://github.com/rust-mobile/ndk/actions/runs/9149848316/job/25153929797

@bbqsrc is it possible to yank and fix this?

**Comment by bbqsrc** [comment]
done

**Pull request merged**